R/separate.R

```diff
@@ -158,3 +158,18 @@ list_indices <- function(x, max = 20) {
 
   paste(x, collapse = ", ")
 }
+
+str_pieces <- function(value, sep, n_max, extra) {
+  pieces <- sapply(value, strsplit, split = sep, perl = TRUE)[[1]]
```

Comment:
Something seems weird here. It seems like this will only work when `value` is length 1?

Reply:
I thought because we are calling via `lapply()` it would only need to handle one string at a time. The only time I could see otherwise is when there is a nested column being passed but I don't think it was handled before.

Running the below on both the current release gave me the same output (reprex from release version, using a data example from nest docs);
``` r
library(dplyr, quietly = TRUE, warn.conflicts = FALSE)
library(tidyr, quietly = TRUE)

df <- tibble(x = c(1, 1, 2, 2), y = c("a.b", "c.d", "d.e", "f.g"))
df %>% nest(data = c(y)) %>%
  separate(data, c("A", "B"), extra = "drop")
#> # A tibble: 2 x 3
#>       x A     B    
#>   <dbl> <chr> <chr>
#> 1     1 list  y    
#> 2     2 list  y
```

Reply:
I think the vectorisation should happen inside of `str_pieces()` not outside of it.

Reply:
Have rewritten to accept vectors and be more straightforward